### PR TITLE
Add dispute types and change InclusionInherent to ParasInherent

### PR DESCRIPTION
**Pull request by rphmeier**
I did the guide update for this ages ago. This makes the chain forwards-compatible for when we add disputes, so we don't have to change the inherent data type later.

### CODE REVIEW

File: primitives/src/v1.rs [hunk 1]
```diff
@@ -991,6 +991,84 @@ impl<H> From<ConsensusLog> for runtime_primitives::DigestItem<H> {
...
+
+/// A set of statements about a specific candidate.
+#[derive(Encode, Decode, Clone)]
+pub struct DisputeStatementSet {
+	/// The candidate referenced by this set.
+    pub candidate_hash: CandidateHash,
+	/// The session index of the candidate.
+    pub session: SessionIndex,
+	/// Statements about the candidate.
+    pub statements: Vec<(DisputeStatement, ValidatorIndex, ValidatorSignature)>,
```
Comment (pepyakin) [inline primitives/src/v1.rs:1051]: spaces

File: runtime/parachains/src/paras_inherent.rs [hunk 2]
```diff
@@ -49,7 +50,7 @@ const MINIMAL_INCLUSION_INHERENT_WEIGHT: Weight = INCLUSION_INHERENT_CLAIMED_WEI
 pub trait Config: inclusion::Config + scheduler::Config {}
 
 decl_storage! {
-	trait Store for Module<T: Config> as ParaInclusionInherent {
+	trait Store for Module<T: Config> as ParaInherent {
```
Comment (pepyakin) [inline runtime/parachains/src/paras_inherent.rs:53]: below some mentions of the "inclusion inherent"

**Review by pepyakin — COMMENTED** [review]
Looks good. I appreciate the renaming and the struct to wrap the para inherent params.

**Comment by rphmeier** [comment]
bot merge

File: primitives/src/v1.rs [hunk 3]
```diff
@@ -22,7 +22,7 @@ use parity_scale_codec::{Encode, Decode};
 use bitvec::vec::BitVec;
 
 use primitives::RuntimeDebug;
-use runtime_primitives::traits::AppVerify;
+use runtime_primitives::traits::{AppVerify, Header as HeaderT};
```
Comment (ordian) [inline primitives/src/v1.rs:25]: naming nit: from what I've seen, the `T` suffix is what we use for generics

**Review by pepyakin — APPROVED** [review]

File: primitives/src/v1.rs (hunk 3 above)
Comment (bkchr) [reply primitives/src/v1.rs:25]: We do this everywhere in Substrate/Polkadot that way.

**Pull request merged**